os/arch/arm/src/armv7-a/arm_assert.c

```diff
@@ -610,7 +614,14 @@ void up_assert(const uint8_t *filename, int lineno)
 	} else {
 		asserted_location = kernel_assert_location;
 	}
-
+#ifdef CONFIG_CRASH_LOG_SAVE
+#ifdef CONFIG_BINMGR_RECOVERY
+	if (IS_FAULT_IN_USER_SPACE(asserted_location)) {
+		/* start assert log saving */
+		set_store_to_buffer_flag(1);
```

Comment:
the fault handler start point might not be here. like `arm_dataabort.c` 
the fault function also have log, the info matters

so in my opinion, we need to set default save on, and we need to stop and start between specific section menually